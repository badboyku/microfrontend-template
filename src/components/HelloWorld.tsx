import * as React from 'react';

type Props = {};

const HelloWorld = (_props: Props): JSX.Element => {
  return <div>Hello World!</div>;
};

export default HelloWorld;
