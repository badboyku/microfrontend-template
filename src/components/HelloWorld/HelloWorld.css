.hello-world {
  color: #7d61fb;
}

.logo {
  width: 40px;
}
