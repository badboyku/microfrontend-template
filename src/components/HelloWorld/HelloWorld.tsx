import * as React from 'react';
import './HelloWorld.css';
import logoUrl, {ReactComponent as Logo} from './logo.svg';

type Props = {};

const HelloWorld = (_props: Props): JSX.Element => {
  return (
    <div>
      <div className="hello-world">Hello World!!!</div>
      <img src={logoUrl} className="logo" alt="logo" />
      <Logo width={40} />
    </div>
  );
};

export default HelloWorld;
