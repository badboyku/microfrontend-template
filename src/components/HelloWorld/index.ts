/* istanbul ignore file */
import HelloWorld from './HelloWorld';

export default HelloWorld;
