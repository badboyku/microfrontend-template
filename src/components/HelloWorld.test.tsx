import * as React from 'react';
import { render, screen } from '@testing-library/react';
import HelloWorld from './HelloWorld';

const renderHelloWorld = (_props = {}) => {
  return render(<HelloWorld />);
};

describe('Component HelloWorld', () => {
  it('renders without crashing', () => {
    const { container } = renderHelloWorld({});
    expect(container).toMatchSnapshot();
  });
  it('renders div with "Hello World!"', () => {
    const { container } = renderHelloWorld({});
    // Has an extra div to serve as the "parent" div for the component
    expect(container).toContainHTML('<div><div>Hello World!</div></div>');
  });
  it('renders one div', () => {
    renderHelloWorld({});
    const helloDiv = screen.getByText('Hello World!');
    expect(helloDiv).toBeInTheDocument();
  });
});
