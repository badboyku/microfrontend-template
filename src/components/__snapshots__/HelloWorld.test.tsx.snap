// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component HelloWorld renders without crashing 1`] = `
<div>
  <div>
    Hello World!
  </div>
</div>
`;
